### ISSUE TITLE
Socket hangup while using STARTTLS

### ISSUE THREAD

**Issue by ambarish-bakre** [mscdex/node-imap#677]
Hi,
I've currently setup a local mailserver(Hmailserver) with IMAP security type to STARTTLS(required).
My IMAP config is 
` "host": "localhost",
                "port": "143",
                "tls": false,
                "autotls": "always",
                "tlsOptions": {
                    "rejectUnauthorized": false
                },
                "debug": false`
I'm getting a socket hangup error when trying to connect to the mailserver.
After a socket has been _upgraded_  to TLS using starttls at
https://github.com/mscdex/node-imap/blob/master/lib/Connection.js#L1646
the library again checks if server supports STARTTLS(which returns true) and again calls `_starttls()` which in turn again creates a new TLS connection and then hangs up because there is already a TLS connection.

As a workaround, I've made the following change at /lib/Connection.js#L1646
`if (self._sock.constructor.name !== 'TLSSocket' && self.serverSupports('STARTTLS')
        && (self._config.autotls === 'always'
          || (self._config.autotls === 'required'
            && self.serverSupports('LOGINDISABLED')))) {
        self._starttls();
        return;
      }`
Basically, `self._sock.constructor.name !== 'TLSSocket'` check would stop the starttls check if our connection is already TLS.
